src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopySendTemporalDataTypesAsStringTest.java

```diff
@@ -0,0 +1,225 @@
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testSendTemporalDataTypesAsStringForBulkCopyDS() throws SQLException {
+        beforeEachSetup();
+        SQLServerDataSource dsLocal = new SQLServerDataSource();
+        AbstractTest.updateDataSource(connectionString, dsLocal);
+        dsLocal.setSendTemporalDataTypesAsStringForBulkCopy(false);
+
+        try (Connection conn = dsLocal.getConnection()) {
+            SQLServerBulkCSVFileRecord fileRecord = new SQLServerBulkCSVFileRecord(filePath + inputFile, encoding,
+                    delimiter, true);
+
+            testBulkCopyCSV(conn, fileRecord);
+        }
+    }
+
+    private void testBulkCopyCSV(Connection conn, SQLServerBulkCSVFileRecord fileRecord) {
+        try (SQLServerBulkCopy bulkCopy = new SQLServerBulkCopy(conn); Statement stmt = conn.createStatement()) {
+
+            fileRecord.addColumnMetadata(1, "id", java.sql.Types.INTEGER, 0, 0);
+            fileRecord.addColumnMetadata(2, "c1", java.sql.Types.DATE, 0, 0); // with Date
+            fileRecord.addColumnMetadata(3, "c2", java.sql.Types.TIMESTAMP, 0, 0); // with Datetime
+            fileRecord.addColumnMetadata(4, "c3", java.sql.Types.TIMESTAMP, 0, 7); // with Datetime2
+            fileRecord.addColumnMetadata(5, "c4", java.sql.Types.TIME, 0, 7); // with time
+            fileRecord.addColumnMetadata(6, "c5", microsoft.sql.Types.DATETIMEOFFSET, 0, 7); // with datetimeoffset
+            fileRecord.addColumnMetadata(7, "c6", java.sql.Types.TIMESTAMP, 0, 0); // with SmallDatetime
+            fileRecord.addColumnMetadata(8, "c7", java.sql.Types.DECIMAL, 19, 4); // with money
+            fileRecord.addColumnMetadata(9, "c8", java.sql.Types.DECIMAL, 10, 4); // with smallmoney
+
+            bulkCopy.setDestinationTableName(destTableName);
+            bulkCopy.writeToServer(fileRecord);
+
+            validateValuesFromCSV(stmt, destTableName, inputFile);
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    private void testBulkCopyResultSet(Connection conn, SQLServerBulkCSVFileRecord fileRecord) {
+        try (SQLServerBulkCopy bulkCopy = new SQLServerBulkCopy(conn); Statement stmt = conn.createStatement()) {
+
+            fileRecord.addColumnMetadata(1, "id", java.sql.Types.INTEGER, 0, 0);
+            fileRecord.addColumnMetadata(2, "c1", java.sql.Types.DATE, 0, 0); // with Date
+            fileRecord.addColumnMetadata(3, "c2", java.sql.Types.TIMESTAMP, 0, 0); // with Datetime
+            fileRecord.addColumnMetadata(4, "c3", java.sql.Types.TIMESTAMP, 0, 7); // with Datetime2
+            fileRecord.addColumnMetadata(5, "c4", java.sql.Types.TIME, 0, 7); // with time
+            fileRecord.addColumnMetadata(6, "c5", microsoft.sql.Types.DATETIMEOFFSET, 0, 7); // with datetimeoffset
+            fileRecord.addColumnMetadata(7, "c6", java.sql.Types.TIMESTAMP, 0, 0); // with SmallDatetime

```

Comment:
The test fails when I switch to:
`fileRecord.addColumnMetadata(7, "c6", microsoft.sql.Types.SMALLDATETIME, 0, 0); // with SmallDatetime`

Reply:
It's supposed to fail. When performing bulk copy, the users are expected to provide the correct column type. The corresponding column type for smalldatetime is TIMESTAMP, and microsoft.sql.Types.SMALLDATETIME is not a supported column type. If you were to test that scenario without my changes, it will fail as well. The table that outlines which datatypes need to be specified for each datatype for bulk copy is outlined here: https://docs.microsoft.com/en-us/sql/connect/jdbc/using-basic-data-types?view=sql-server-2017

Reply:
Understood.